src/mol-gl/shader/cylinders.vert.ts

```diff
@@ -50,6 +54,7 @@ void main() {
     vEnd = (modelTransform * vec4(aEnd, 1.0)).xyz;
     vSize = size * aScale;
     vCap = aCap;
+    vColorMode = aColorMode;
```

Comment:
I think you can pre-mix vColor and vColor2 here already instead of in the frag shader, i.e., set vColor and vColor2 to values here in a way you don't need a branch when mixing/interpolating them in the frag shader.

Reply:
Hmm.. I'm not sure what you mean here.

Reply:
went ahead and did it in da034d950269f28748a76b1d40f23fcd3c8fe7c9